### PR TITLE
feat: add log level to flags

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

Add log level to the flags page

## Notice

- [ ] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Risk

no risk

## Testing

N/A

## Screenshots (if applicable)
Screen Shot 2022-05-06 at 3 38 42 PM

### CODE REVIEW

File: src/lib/logger.ts [hunk 1]
```diff
@@ -52,8 +55,168 @@ const browserLoggerFn: LoggerFunction = (level, data) => {
...
+    }
+  }
+
+  return false
+}
+
+export const logger = createLogger()
+
+/*
+ DOCUMENATION
```
Comment (0xApotheosis) [inline src/lib/logger.ts:111]:
```suggestion
 DOCUMENTATION
```

File: src/lib/logger.ts [hunk 2]
```diff
@@ -52,8 +55,168 @@ const browserLoggerFn: LoggerFunction = (level, data) => {
...
+
+ # Why
+ The purpose of this logger is to allow us to log out information in a structured way that
+ makes debugging easier by allowing us to more easily filter and search logs.
```
Comment (0xApotheosis) [inline src/lib/logger.ts:115]: Great docs, though I feel like they should be in a README somewhere?

File: src/pages/Flags/Debugging.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                      if (typeof value === 'string') saveLogLevel(value)
+                      setLogLevel(getLogLevel())
+                    }}
+                  >
+                    <MenuItemOption value='error'>Error</MenuItemOption>
+                    <MenuItemOption value='warn'>Warning</MenuItemOption>
+                    <MenuItemOption value='info'>Info</MenuItemOption>
+                    <MenuItemOption value='debug'>Debug</MenuItemOption>
+                    <MenuItemOption value='trace'>Trace</MenuItemOption>
+                    <MenuItemOption value='none'>None</MenuItemOption>
```
Comment (0xApotheosis) [inline src/pages/Flags/Debugging.tsx:47]: It's a little weird clicking on a sentence case label and the newly selected value then being lower case.

**Review by 0xApotheosis — APPROVED** [review]
A few non-blocking comments.

File: src/lib/logger.ts (hunk 2 above)
Comment (cjthompson) [reply src/lib/logger.ts:115]: I asked def1 about this and he suggested that most developers would come across this code and its comment before looking at a file somewhere else. 

I will probably also make a PR to the Docs repo. If so I'll link that from here.

**Review by 0xApotheosis — APPROVED** [review]

**Review by 0xdef1cafe — APPROVED** [review]

**Comment by 0xdef1cafe** [comment]
LGTM - nice addition

**Pull request merged**